<a href="https://colab.research.google.com/github/hyeyeonIm/SJSU_DATA/blob/main/Copy_of_CS22A_Term_Project_Datasets.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## **Choose one of the following datasets for your CS 22A term project**

If you have another dataset that you want to use, please send it to your instructor for approval.


##**US Health Insurance**

<img src='https://roundtmc.com/wp-content/uploads/2017/09/health-insurance.jpg' width=660>

This dataset contains 1338 rows of insured data, where the Insurance charges are given against the following attributes of the insured: Age, Sex, BMI, Number of Children, Smoker and Region.

Here are the descriptions of this dataset:
- `age` - age of this individual
- `sex` - gender
- `bmi` - body mass index
- `children` - number of children this individual has
- `smoker` - smoke regularly or not
- `region` - the US region in which this individual resides
- `charges` - the amount of health insurance charges for this indivudal

In [ ]:
import pandas as pd
data_url = 'https://raw.githubusercontent.com/csbfx/cs22a/main/us_health_insurance.csv'
ins_df = pd.read_csv(data_url)
ins_df

age     sex     bmi  children smoker     region      charges
0      19  female  27.900         0    yes  southwest  16884.92400
1      18    male  33.770         1     no  southeast   1725.55230
2      28    male  33.000         3     no  southeast   4449.46200
3      33    male  22.705         0     no  northwest  21984.47061
4      32    male  28.880         0     no  northwest   3866.85520
...   ...     ...     ...       ...    ...        ...          ...
1333   50    male  30.970         3     no  northwest  10600.54830
1334   18  female  31.920         0     no  northeast   2205.98080
1335   18  female  36.850         0     no  southeast   1629.83350
1336   21  female  25.800         0     no  southwest   2007.94500
1337   61  female  29.070         0    yes  northwest  29141.36030

[1338 rows x 7 columns]

---

##**Pokemon**

<img src="https://suntrustblog.com/the-best-pokemon-games/maxresdefault-62/" width=700>

This dataset is about the pokemon games (NOT pokemon cards or Pokemon Go). It includes 721 Pokemon, including their number, name, first and second type, and basic stats: HP, Attack, Defense, Special Attack, Special Defense, and Speed. These are the raw attributes used for calculating how much damage an attack will do in the games.

Here are the descriptions of this dataset:
- `#` - Pokedex entry number of the Pokemon
- `Name` - name of the Pokemon
- `Type 1` - each Pokemon has a type, this determines weakness/resistance to attacks [referred to as the primary type]
- `Type 2` - some Pokemon are dual type and have 2 [referred to as the secondary type]
- `Total` - sum of all stats that come after this, a general guide to how strong a Pokemon is
- `HP` - hit points, or health, defines how much damage a Pokemon can withstand before fainting
- `Attack` - the base modifier for normal attacks
- `Defense` - the base damage resistance against normal attacks
- `Sp. Atk` - special attack, the base modifier for special attacks
- `Sp. Def` - the base damage resistance against special attacks
- `Speed` - determines which Pokemon attacks first each round
- `Generation` - refers to which grouping/game series the Pokemon was released in
- `Legendary` - a boolean that identifies whether the Pokemon is legendary

In [ ]:
import pandas as pd

data_url = 'https://raw.githubusercontent.com/csbfx/advpy122-data/master/Pokemon.csv'
poke_df = pd.read_csv(data_url)
poke_df

#                   Name   Type 1  Type 2  Total  HP  Attack  Defense  \
0      1              Bulbasaur    Grass  Poison    318  45      49       49   
1      2                Ivysaur    Grass  Poison    405  60      62       63   
2      3               Venusaur    Grass  Poison    525  80      82       83   
3      3  VenusaurMega Venusaur    Grass  Poison    625  80     100      123   
4      4             Charmander     Fire     NaN    309  39      52       43   
..   ...                    ...      ...     ...    ...  ..     ...      ...   
795  719                Diancie     Rock   Fairy    600  50     100      150   
796  719    DiancieMega Diancie     Rock   Fairy    700  50     160      110   
797  720    HoopaHoopa Confined  Psychic   Ghost    600  80     110       60   
798  720     HoopaHoopa Unbound  Psychic    Dark    680  80     160       60   
799  721              Volcanion     Fire   Water    600  80     110      120   

     Sp. Atk  Sp. Def  Speed  Generation  Legendary  
0         65       65     45           1      False  
1         80       80     60           1      False  
2        100      100     80           1      False  
3        122      120     80           1      False  
4         60       50     65           1      False  
..       ...      ...    ...         ...        ...  
795      100      150     50           6       True  
796      160      110    110           6       True  
797      150      130     70           6       True  
798      170      130     80           6       True  
799      130       90     70           6       True  

[800 rows x 13 columns]

---



##**Heart Disease**

<img src="https://domf5oio6qrcr.cloudfront.net/medialibrary/10112/heart-and-ekg.jpg" width=660>  

The term "heart disease" is often used interchangeably with the term "cardiovascular disease". Cardiovascular disease generally refers to conditions that involve narrowed or blocked blood vessels that can lead to a heart attack, chest pain (angina) or stroke. Other heart conditions, such as those that affect your heart’s muscle, valves or rhythm, also are considered forms of heart disease.

This dataset gives a number of variables of individuals with or without heart disease.

- `age`: The person's age in years
- `sex`: The person's sex (1 = male, 0 = female)
- `chest_pain`: The chest pain experienced (Value 1: typical angina, Value 2: atypical angina, Value 3: non-anginal pain, Value 4: asymptomatic)
- `resting_bp`: The person's resting blood pressure (mm Hg on admission to the hospital)
- `cholesterol`: The person's cholesterol measurement in mg/dl
- `fasting_blood_sugar`: The person's fasting blood sugar (> 120 mg/dl, 1 = true; 0 = false)
- `rest_ecg`: Resting electrocardiographic measurement (0 = normal, 1 = having ST-T wave abnormality, 2 = showing probable or definite left ventricular hypertrophy by Estes' criteria)
- `thalach`: The person's maximum heart rate achieved
- `exang`: Exercise induced angina (1 = yes; 0 = no)
- `oldpeak`: ST depression induced by exercise relative to rest ('ST' relates to positions on the ECG plot. See more here)
- `slope`: the slope of the peak exercise ST segment (Value 1: upsloping, Value 2: flat, Value 3: downsloping)
ca: The number of major vessels (0-3)
- `thalassemia`: A blood disorder called thalassemia (3 = normal; 6 = fixed defect; 7 = reversable defect)
- `heart_disease`: Heart disease (0 = no, 1 = yes)

In [ ]:
import pandas as pd

data_url = 'https://raw.githubusercontent.com/csbfx/cs22a/main/heart.csv'
heart_df = pd.read_csv(data_url)
heart_df

age  sex  chest_pain  resting_bp  cholesterol  fasting_blood_sugar  \
0     63    1           3         145          233                    1   
1     37    1           2         130          250                    0   
2     41    0           1         130          204                    0   
3     56    1           1         120          236                    0   
4     57    0           0         120          354                    0   
..   ...  ...         ...         ...          ...                  ...   
298   57    0           0         140          241                    0   
299   45    1           3         110          264                    0   
300   68    1           0         144          193                    1   
301   57    1           0         130          131                    0   
302   57    0           1         130          236                    0   

     rest_ecg  thalachh  exng  oldpeak  slope  num_major_vessels  thalassemia  \
0           0       150     0      2.3      0                  0            1   
1           1       187     0      3.5      0                  0            2   
2           0       172     0      1.4      2                  0            2   
3           1       178     0      0.8      2                  0            2   
4           1       163     1      0.6      2                  0            2   
..        ...       ...   ...      ...    ...                ...          ...   
298         1       123     1      0.2      1                  0            3   
299         1       132     0      1.2      1                  0            3   
300         1       141     0      3.4      1                  2            3   
301         1       115     1      1.2      1                  1            3   
302         0       174     0      0.0      1                  1            2   

     heart_disease  
0                1  
1                1  
2                1  
3                1  
4                1  
..             ...  
298              0  
299              0  
300              0  
301              0  
302              0  

[303 rows x 14 columns]

---

##**Global Climate Change**

<img src='https://static1.squarespace.com/static/59a706d4f5e2319b70240ef9/t/5a7a3f018165f5e5566ca3e6/1517961006265/AdobeStock_54132491.jpeg' width=700>

In recent years, we already see effects scientists predicted, such as the loss of sea ice, melting glaciers and ice sheets, sea level rise, and more intense heat waves. Scientists predict global temperature increases from human-made greenhouse gases will continue. Severe weather damage will also increase and intensify.


This is a dataset with average temperatures in Fahrenheit for 157 U.S. and 167 international cities measured from 1995 to 2020 from the [Global Summary of the Day (GSOD)](https://www.ncei.noaa.gov/access/metadata/landing-page/bin/iso?id=gov.noaa.ncdc:C00516) database.

Here are the descriptions of the dataset:

- `Region` - Global region
- `Country` - Country
- `City` - City
- `Year` - Year the temperatures were measured
- `AvgTemperature` - Average temperature (°F)

In [ ]:
import pandas as pd
data_url = 'https://raw.githubusercontent.com/csbfx/cs22a/main/global_climate.csv'

climate_df = pd.read_csv(data_url)
climate_df

Region    Country     City  Year  \
0                                Africa    Algeria  Algiers  1995   
1                                Africa    Algeria  Algiers  1996   
2                                Africa    Algeria  Algiers  1997   
3                                Africa    Algeria  Algiers  1998   
4                                Africa    Algeria  Algiers  1999   
...                                 ...        ...      ...   ...   
7959  South/Central America & Carribean  Venezuela  Caracas  2016   
7960  South/Central America & Carribean  Venezuela  Caracas  2017   
7961  South/Central America & Carribean  Venezuela  Caracas  2018   
7962  South/Central America & Carribean  Venezuela  Caracas  2019   
7963  South/Central America & Carribean  Venezuela  Caracas  2020   

      AvgTemperature  
0          64.410468  
1          63.530220  
2          64.926301  
3          63.619668  
4          64.923626  
...              ...  
7959       81.707055  
7960       82.844575  
7961       79.969714  
7962       79.816715  
7963       80.261654  

[7964 rows x 5 columns]

---

##**S&P 500 Companies**

<img src="https://t3.ftcdn.net/jpg/02/78/47/86/360_F_278478658_3VTuToNWgIsM81lAtrFuag3u2vrbuFbn.jpg" width=660>  

The Standard and Poor's 500 or S&P 500 is the most famous financial benchmark in the world. This stock market index tracks the performance of 500 large companies listed on stock exchanges in the United States. As of December 31, 2020, more than $5.4 trillion was invested in assets tied to the performance of this index.

Here are the descriptions of this dataset:

- `Exchange`: Exchange where its stocks are negotiated
- `Symbol`: Stock symbol
- `Shortname`: Company short name
- `Longname`: Company long name
- `Sector`: Sector where the company operates
- `Industry`: Industry, within a sector, where the company operates.
- `Currentprice`: Current stock price
- `Marketcap`: Current marketcap
- `Ebitda`: Earnings before interest, taxes, depreciation and amortization
- `Revenuegrowth`: Revenue growth
- `City`: City
- `State`: State
- `Country`: Country
- `Fulltimeemployees`: Number of full time employees
- `Longbusinesssummary`: Summary of the company
- `Weight`: Percentage of participation on the S&P index (marketcap %)

In [ ]:
import pandas as pd
data_url = 'https://raw.githubusercontent.com/csbfx/cs22a/main/sp500_companies.csv'

sp500_df = pd.read_csv(data_url)
sp500_df

Exchange Symbol                Shortname                 Longname  \
0        NMS   AAPL               Apple Inc.               Apple Inc.   
1        NMS   MSFT    Microsoft Corporation    Microsoft Corporation   
2        NMS   GOOG            Alphabet Inc.            Alphabet Inc.   
3        NMS  GOOGL            Alphabet Inc.            Alphabet Inc.   
4        NMS   AMZN         Amazon.com, Inc.         Amazon.com, Inc.   
..       ...    ...                      ...                      ...   
489      NYQ    OGN            Organon & Co.            Organon & Co.   
490      NYQ    MHK  Mohawk Industries, Inc.  Mohawk Industries, Inc.   
491      NMS    NWL       Newell Brands Inc.       Newell Brands Inc.   
492      NYQ    ALK   Alaska Air Group, Inc.   Alaska Air Group, Inc.   
493      NYQ    VNO     Vornado Realty Trust     Vornado Realty Trust   

                     Sector                            Industry  Currentprice  \
0                Technology                Consumer Electronics        155.74   
1                Technology             Software—Infrastructure        235.87   
2    Communication Services      Internet Content & Information         96.58   
3    Communication Services      Internet Content & Information         96.29   
4         Consumer Cyclical                     Internet Retail        103.41   
..                      ...                                 ...           ...   
489              Healthcare          Drug Manufacturers—General         26.18   
490       Consumer Cyclical  Furnishings, Fixtures & Appliances         96.72   
491      Consumer Defensive       Household & Personal Products         15.05   
492             Industrials                            Airlines         44.43   
493             Real Estate                         REIT—Office         23.80   

         Marketcap        Ebitda  Revenuegrowth           City State  \
0    2502866501632  1.305410e+11          0.081      Cupertino    CA   
1    1770907107328  9.798300e+10          0.124        Redmond    WA   
2    1248045432832  9.373300e+10          0.061  Mountain View    CA   
3    1248043597824  9.373300e+10          0.061  Mountain View    CA   
4    1054957895680  5.154900e+10          0.147        Seattle    WA   
..             ...           ...            ...            ...   ...   
489     6637939200  2.100000e+09          0.041    Jersey City    NJ   
490     6551077888  1.888004e+09          0.067        Calhoun    GA   
491     6402270208  1.390000e+09         -0.065        Atlanta    GA   
492     5635234816  1.073000e+09          0.448        Seattle    WA   
493     4562007552  8.103430e+08          0.163       New York    NY   

           Country  Fulltimeemployees  \
0    United States           164000.0   
1    United States           221000.0   
2    United States           186779.0   
3    United States           186779.0   
4    United States          1544000.0   
..             ...                ...   
489  United States             9300.0   
490  United States            43000.0   
491  United States            32000.0   
492  United States            22354.0   
493  United States             3224.0   

                                   Longbusinesssummary    Weight  
0    Apple Inc. designs, manufactures, and markets ...  0.072117  
1    Microsoft Corporation develops, licenses, and ...  0.051026  
2    Alphabet Inc. provides various products and pl...  0.035961  
3    Alphabet Inc. provides various products and pl...  0.035961  
4    Amazon.com, Inc. engages in the retail sale of...  0.030397  
..                                                 ...       ...  
489  Organon & Co., a health care company, develops...  0.000191  
490  Mohawk Industries, Inc. designs, manufactures,...  0.000189  
491  Newell Brands Inc. designs, manufactures, sour...  0.000184  
492  Alaska Air Group, Inc., through its subsidiari...  0.000162  
493  Vornado's portfolio is concentrated in the nat...  0.0

---

##**Fitness Product Market Research**

<img src="https://www.treadmillreviews.net/wp-content/uploads/2020/03/treadmillrunners.png" width=660>

A market research team is assigned the task to identify the profile of the typical customer for each treadmill product offered by Fitness for Everyone. The market research team decides to investigate whether there are differences across the product lines with respect to customer characteristics. The team decides to collect data on individuals who purchased a treadmill at a Fitness for Everyone retail store during the prior three months. The team identifies the following customer variables to study:

- `Product`: Product purchased - FE285, FE580, or FE605
- `Age`: Age in years
- `gender`:  Male/Female
- `Education`: Years of education received
- `MaritalStatus`: relationship status - single or partnered
- `Usage`: Average number of times the customer plans to use the treadmill each week
- `Fitness`: Self-rated fitness on an 1-to-5 scale, where 1 is poor shape and 5 is excellent shape.  
- `Income`: Annual household income ($)
- `Miles`: Average number of miles the customer expects to walk/run each week


In [ ]:
import pandas as pd

data_url = 'https://raw.githubusercontent.com/csbfx/cs22a/main/fitness.csv'

fitness_df = pd.read_csv(data_url)
fitness_df

Product  Age  Gender  Education MaritalStatus  Usage  Fitness  Income  \
0     FE284   18    Male         14        Single      3        4   29562   
1     FE284   19    Male         15        Single      2        3   31836   
2     FE284   19  Female         14     Partnered      4        3   30699   
3     FE284   19    Male         12        Single      3        3   32973   
4     FE284   20    Male         13     Partnered      4        2   35247   
..      ...  ...     ...        ...           ...    ...      ...     ...   
175   FE605   40    Male         21        Single      6        5   83416   
176   FE605   42    Male         18        Single      5        4   89641   
177   FE605   45    Male         16        Single      5        5   90886   
178   FE605   47    Male         18     Partnered      4        5  104581   
179   FE605   48    Male         18     Partnered      4        5   95508   

     Miles  
0      112  
1       75  
2       66  
3       85  
4       47  
..     ...  
175    200  
176    200  
177    160  
178    120  
179    180  

[180 rows x 9 columns]

---

##**Diabetes**

<img src="https://www.niddk.nih.gov/-/media/Images/Health-Information/Diabetes/diabetes-monitor-fruits-vegetables-small_597x347.png" width=660>  

Diabetes is among the most prevalent chronic diseases in the United States, impacting millions of Americans each year and exerting a significant financial burden on the economy. Diabetes is a serious chronic disease in which individuals lose the ability to effectively regulate levels of glucose in the blood, and can lead to reduced quality of life and life expectancy.

The Behavioral Risk Factor Surveillance System (BRFSS) is a health-related telephone survey that is collected annually by the CDC. Each year, the survey collects responses from over 400,000 Americans on health-related risk behaviors, chronic health conditions, and the use of preventative services. It has been conducted every year since 1984. This dataset is a randomly selection of 5,000 surveys collected from 2015.

Here are the descriptions of the data:

- `Diabetes_012`: diagnosis, `0` = no diabetes, `1` = prediabetes, `2` = diabetes
- `HighBP`: high blood pressure, `0` = no high, BP `1` = high BP
- `HighChol`: high cholesterol, `0` = no high cholesterol, `1` = high cholesterol
- `CholCheck`: cholesterol check, `0` = no cholesterol check in 5 years, `1` = yes cholesterol check in 5 years
- `BMI`: Body Mass Index
- `Smoker`: smoked at least 100 cigarettes in your entire life, `0` = no, `1` = yes
- `Stroke`: had a stroke before, `0` = no, `1` = yes
- `HeartDiseaseorAttack`: coronary heart disease (CHD) or myocardial infarction (MI), `0` = no, `1` = yes
- `PhysActivity`: physical activity in past `30` days - not including job, `0` = no, `1` = yes
- `Fruits`: Consume Fruit 1 or more times per day, `0` = no, `1` = yes
- `Veggies`: Consume Vegetables 1 or more times per day, `0` = no, `1` = yes
- `HvyAlcoholConsump`: Heavy drinkers (adult men having more than 14 drinks per week and adult women having more than `7` drinks per week), `0` = no, `1` = yes
- `AnyHealthcare`: Have any kind of health care coverage, including health insurance, prepaid plans such as HMO, etc. `0` = no, `1` = yes
- `GenHlth`: self-evaluation of health - scale `1-5`, `1` = excellent, `2` = very good, `3` = good, `4` = fair, `5` = poor
- `DiffWalk`: Have serious difficulty walking or climbing stairs, `0` = no, `1` = yes
- `Sex`: Gender, `0` = female, `1` = male
- `Age`: 13-level age category, `1` = `18-24`, `9` = `60-64`, `13` = `80 or older`
- `Education`: Education level, scale `1-6`, `1` = Never attended school or only kindergarten, `2` = Grades 1 through 8 (Elementary), `3` = Grades 9 through 11 (Some high school), `4` = Grade 12 or GED (High school graduate), `5` = College 1 year to 3 years (Some college or technical school), `6` = College 4 years or more (College graduate)
- `Income`: Income scale, scale `1-8`, `1` = less than `$10,000`, `5` = less than `$35,000`, `8` = `$75,000` or more


In [ ]:
import pandas as pd

data_url = 'https://raw.githubusercontent.com/csbfx/cs22a/main/diabetes5k.csv'
ddf = pd.read_csv(data_url)
ddf

Diabetes_012  HighBP  HighChol  CholCheck   BMI  Smoker  Stroke  \
0              0.0     0.0       1.0        1.0  25.0     1.0     0.0   
1              0.0     0.0       0.0        1.0  28.0     0.0     0.0   
2              0.0     0.0       0.0        1.0  25.0     1.0     0.0   
3              0.0     0.0       1.0        1.0  27.0     0.0     0.0   
4              0.0     0.0       0.0        1.0  27.0     0.0     0.0   
...            ...     ...       ...        ...   ...     ...     ...   
4995           0.0     0.0       0.0        1.0  21.0     0.0     0.0   
4996           0.0     1.0       1.0        1.0  22.0     0.0     0.0   
4997           0.0     0.0       1.0        1.0  26.0     0.0     0.0   
4998           0.0     1.0       0.0        1.0  41.0     0.0     0.0   
4999           0.0     0.0       0.0        1.0  26.0     0.0     0.0   

      HeartDiseaseorAttack  PhysActivity  Fruits  Veggies  HvyAlcoholConsump  \
0                      0.0           1.0     0.0      1.0                0.0   
1                      0.0           1.0     1.0      1.0                0.0   
2                      0.0           1.0     0.0      1.0                0.0   
3                      0.0           1.0     1.0      1.0                0.0   
4                      0.0           1.0     1.0      0.0                0.0   
...                    ...           ...     ...      ...                ...   
4995                   0.0           0.0     0.0      1.0                0.0   
4996                   0.0           1.0     0.0      1.0                0.0   
4997                   0.0           0.0     0.0      1.0                0.0   
4998                   0.0           1.0     0.0      1.0                0.0   
4999                   0.0           0.0     1.0      1.0                0.0   

      AnyHealthcare  GenHlth  DiffWalk  Sex   Age  Education  Income  
0               1.0      2.0       0.0  0.0   7.0        5.0     8.0  
1               1.0      2.0       0.0  1.0   6.0        6.0     8.0  
2               0.0      3.0       0.0  0.0   7.0        4.0     6.0  
3               1.0      2.0       0.0  1.0   9.0        5.0     8.0  
4               1.0      2.0       0.0  0.0   7.0        4.0     3.0  
...             ...      ...       ...  ...   ...        ...     ...  
4995            1.0      1.0       0.0  0.0   3.0        6.0     3.0  
4996            1.0      3.0       0.0  0.0   9.0        5.0     8.0  
4997            1.0      2.0       0.0  1.0   8.0        5.0     8.0  
4998            1.0      3.0       0.0  0.0   6.0        5.0     5.0  
4999            1.0      2.0       0.0  0.0  11.0        4.0     3.0  

[5000 rows x 19 columns]

---

##**Most Popular Dog Breeds**

<img src="https://tipsclear.com/wp-content/uploads/2020/04/dog-breeds.jpg" width=660>  

Like clothing styles and baby names, dog breeds go in and out of style. It’s likely that pop culture plays a large role in preferences. In fact, a study from the University of Bristol, City University of New York, and Western Carolina University found that movies, specifically durably popular ones, play a large role in breed popularity. For most of the twentieth century, even the breed’s temperament, health, and longevity played less of a role in selecting a dog than did association with a popular film.

Here are the data descriptions

- `AKC_group`: American Kennel Club group
- `Popularity_in_US`: 1- 173, 1 = most poular, US. American Kennel Club registrations, pure breeds only
- `Intelligence`: 1 = easiest to train / learn new commands
- `Longevity`: years, weighted average - see note
- `Genetic_ailments`: serious genetic ailments only
- `Avg_price`: average price from breeders
- `Size`: small, medium, large
- `Weight` (lbs): weight by pound
- `Shoulder_height (in)`: shoulder height in inches
- `Intelligence_category`: Average, Above average, Fair, Excellent,  Lowest, Brightest


In [ ]:
import pandas as pd
data_url = 'https://raw.githubusercontent.com/csbfx/cs22a/main/AKC_dogs_rankings.csv'
dog_df = pd.read_csv(data_url)
dog_df


Breed     AKC_group  Popularity_in_US  \
0                  Border Collie       herding                45   
1                 Border Terrier       terrier                80   
2                       Brittany      sporting                30   
3                  Cairn Terrier       terrier                59   
4         Welsh Springer Spaniel      sporting               130   
..                           ...           ...               ...   
167                   Weimaraner      sporting                32   
168                Welsh Terrier       terrier                99   
169             Wire Fox Terrier       terrier               100   
170  Wirehaired Pointing Griffon      sporting                92   
171               Xoloitzcuintli  non-sporting               155   

     Intelligence Longevity  Genetic_ailments Avg_price    Size  Weight (lbs)  \
0               1     12.52                 2       623  medium          48.0   
1              30     14.00                 0       833   small          13.5   
2              19     12.92                 0       618  medium          35.0   
3              35     13.84                 2       435   small          14.0   
4              31     12.49                 1       750  medium          51.0   
..            ...       ...               ...       ...     ...           ...   
167            21        12                 1       562   large          69.0   
168            53        13                 0       843   small          20.0   
169            51     13.17                 0       668   small          17.5   
170            46      8.80                 0       755  medium          48.0   
171            56        13                 0       717  medium          52.0   

     Shoulder_height (in) Intelligence_category  
0                    20.0             Brightest  
1                    11.0         Above average  
2                    19.0             Excellent  
3                    10.0         Above average  
4                    18.0         Above average  
..                    ...                   ...  
167                  25.0             Excellent  
168                  15.0               Average  
169                  15.0               Average  
170                  22.0               Average  
171                  16.5               Average  

[172 rows x 11 columns]

---

##**Rates of COVID-19 Cases and Deaths by Vaccination Status**

<img src="https://www.coe.int/documents/10518249/88399762/Covid-Vaccine/23edacee-ac47-953e-7c8d-012ec176c157" width=660>

The Vaccination outcomes data track and compare the rates of COVID-19 infections and deaths among unvaccinated and vaccinated people, showing how vaccination reduces the risk of severe outcomes from COVID-19. The following data were posted on October 21, 2022 by the CDC and reflect cases among persons with a positive specimen collection date through September 24, 2022, and deaths among persons with a positive specimen collection date through September 3, 2022.


Vaccination status: A person vaccinated with a primary series had SARS-CoV-2 RNA or antigen detected on a respiratory specimen collected ≥14 days after verifiably completing the primary series of an FDA-authorized or approved COVID-19 vaccine. An unvaccinated person had SARS-CoV-2 RNA or antigen detected on a respiratory specimen and has not been verified to have received COVID-19 vaccine.

Here are the descriptions of the dataset:

- `outcome`: COVID-19 Case or Death
- `month`: Calendar month and year corresponding to MMWR week value [MMM YYYY format; e.g.APR 2021]
- `MMWR week`: MMWR epidemiological year and week [YYYYWW format; e.g. 202101]
- `Age group`: Age Group [0-4 years; 5-1 years; 12-17 years; 18-29 years; 30-49 years; 50-64 years; 65-79 years; 80+ years]
- `Vaccinated with outcome`: Weekly count of individuals vaccinated with at least a primary series with the corresponding outcome
- `Unvaccinated with outcome`: Weekly count of unvaccinated individuals with the corresponding outcome
- `Crude vax IR`: Unadjusted incidence rate of the corresponding outcome among the population vaccinated with at least a primary series (per 100,000 population)
- `Crude unvax IR`: Unadjusted incidence rate of the corresponding outcome among the unvaccinated population (per 100,000 population)


In [ ]:
import pandas as pd
data_url = 'https://raw.githubusercontent.com/csbfx/cs22a/main/covid19_rate_by_vaccine.csv'

cov19_df = pd.read_csv(data_url)
cov19_df

outcome     month  MMWR week Age group  Vaccinated with outcome  \
0      case  APR 2021     202114     12-17                        8   
1      case  APR 2021     202114     18-29                      671   
2      case  APR 2021     202114     30-49                     1837   
3      case  APR 2021     202114     50-64                     1547   
4      case  APR 2021     202114     65-79                     1655   
..      ...       ...        ...       ...                      ...   
982   death  AUG 2022     202235     18-29                        2   
983   death  AUG 2022     202235     30-49                       11   
984   death  AUG 2022     202235     50-64                       46   
985   death  AUG 2022     202235     65-79                      201   
986   death  AUG 2022     202235       80+                      309   

     Unvaccinated with outcome  Crude vax IR  Crude unvax IR  
0                        30785     21.687858      175.348491  
1                        76736     26.385193      246.808299  
2                        98436     24.729400      239.653115  
3                        50324     20.661256      219.290266  
4                        14880     11.494573      193.852934  
..                         ...           ...             ...  
982                          1      0.009043        0.008873  
983                         18      0.027152        0.147408  
984                         67      0.138830        1.113800  
985                        140      0.819404        7.364142  
986                        172      4.365044       17.859082  

[987 rows x 8 columns]